packages/kubernetes-1.20/credential-provider-config.yaml

```
@@ -0,0 +1,26 @@
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: CredentialProviderConfig
+{{! This isn't right as the condition is more than just whether `providers` is "truey" but whether any are enabled }}
+{{#if settings.kubernetes.credential-providers.providers}}
+providers:
+{{#each settings.kubernetes.credential-providers.providers}}
+{{#if this.enabled}}
+  - name: {{this.name}}
+    matchImages:
+{{#each this.imagepatterns}}
+      - {{this}}
+{{/each}}
+    defaultCacheDuration: {{default "12h" this.cacheduration}}
+    apiVersion: credentialprovider.kubelet.k8s.io/v1beta1
```

I'd prefer the usual kebab-case style for the settings fields:
```suggestion
{{#each this.image-patterns}}
      - {{this}}
{{/each}}
    defaultCacheDuration: {{default "12h" this.cache-duration}}
    apiVersion: credentialprovider.kubelet.k8s.io/v1beta1
```